sympy/assumptions/ask.py

```diff
@@ -11,52 +11,479 @@
 from sympy.utilities.decorator import classproperty, ClassPropertyDescriptor
 
 
+# Deprecated predicates should be added to this list
+deprecated_predicates = [
+    'bounded',
+    'infinity'
+]
+
+
 class Q(object):
-    """Supported ask keys."""
-    antihermitian = Predicate('antihermitian')
-    finite = Predicate('finite')
-    commutative = Predicate('commutative')
-    complex = Predicate('complex')
-    composite = Predicate('composite')
-    even = Predicate('even')
-    extended_real = Predicate('extended_real')
-    hermitian = Predicate('hermitian')
-    imaginary = Predicate('imaginary')
-    infinitesimal = Predicate('infinitesimal')
-    infinite = Predicate('infinite')
-    integer = Predicate('integer')
-    irrational = Predicate('irrational')
-    rational = Predicate('rational')
-    algebraic = Predicate('algebraic')
-    transcendental = Predicate('transcendental')
-    negative = Predicate('negative')
-    nonzero = Predicate('nonzero')
-    positive = Predicate('positive')
-    prime = Predicate('prime')
-    real = Predicate('real')
-    odd = Predicate('odd')
-    is_true = Predicate('is_true')
-    nonpositive = Predicate('nonpositive')
-    nonnegative = Predicate('nonnegative')
-    zero = Predicate('zero')
-
-    symmetric = Predicate('symmetric')
-    invertible = Predicate('invertible')
-    singular = Predicate('singular')
-    orthogonal = Predicate('orthogonal')
-    unitary = Predicate('unitary')
-    normal = Predicate('normal')
-    positive_definite = Predicate('positive_definite')
-    upper_triangular = Predicate('upper_triangular')
-    lower_triangular = Predicate('lower_triangular')
-    diagonal = Predicate('diagonal')
-    triangular = Predicate('triangular')
-    unit_triangular = Predicate('unit_triangular')
-    fullrank = Predicate('fullrank')
-    square = Predicate('square')
-    real_elements = Predicate('real_elements')
-    complex_elements = Predicate('complex_elements')
-    integer_elements = Predicate('integer_elements')
+    """
+    This class contains all the supported keys by ``ask``.
+    """
+
+    @classproperty
+    def antihermitian(self):
+        r"""
+        Antihermitian predicate.
+
+        ``ask(Q.antihermitian(x))`` is true iff x belongs to the field of
+        antihermitian operators.
+
+        TODO: Add examples
+        """
+        return Predicate('antihermitian')
+
+    @classproperty
+    def finite(self):
+        r"""
+        Finite number predicate.
```

Comment:
"number" is maybe not the best word here. Some people may not agree whether or not `I` is a "number".